### ISSUE TITLE
DiffusionPipeline does not give informative error messages when a dependency is missing

### ISSUE THREAD

**Issue by osanseviero** [huggingface/diffusers#4598]
### Describe the bug

If I use `StableDiffusionXLPipeline`, I get a good error message if `transformers` is missing. If I use `DiffusionPipeline` instead, the error is not transparent.

### Reproduction

Colab  https://colab.research.google.com/drive/1fF7r-FLg729PbQNAn03179vqWDTkDrTP?usp=sharing

```python
from diffusers import DiffusionPipeline, StableDiffusionXLPipeline
import torch

DiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16)
```

and

```python
StableDiffusionXLPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16)
```

(don't install transformers in the environment)

### Logs
From `DiffusionPipeline`

```
/usr/local/lib/python3.10/dist-packages/diffusers/pipelines/pipeline_utils.py in download(cls, pretrained_model_name, **kwargs)
   1428                 ignore_patterns = ["*.bin", "*.msgpack"]
   1429 
-> 1430                 use_onnx = use_onnx if use_onnx is not None else pipeline_class._is_onnx
   1431                 if not use_onnx:
   1432                     ignore_patterns += ["*.onnx", "*.pb"]

/usr/local/lib/python3.10/dist-packages/diffusers/utils/import_utils.py in __getattr__(cls, key)
    573     def __getattr__(cls, key):
    574         if key.startswith("_") and key != "_load_connected_pipes":
--> 575             return super().__getattr__(cls, key)
    576         requires_backends(cls, cls._backends)
    577 

AttributeError: 'super' object has no attribute '__getattr__'
```

From `StableDiffusionXLPipeline`

```
/usr/local/lib/python3.10/dist-packages/diffusers/utils/import_utils.py in requires_backends(obj, backends)
    542     failed = [msg.format(name) for available, msg in checks if not available()]
    543     if failed:
--> 544         raise ImportError("".join(failed))
    545 
    546     if name in [

ImportError: 
StableDiffusionXLPipeline requires the transformers library but it was not found in your environment. You can install it with pip: `pip
install transformers`
```

### System Info

- `diffusers` version: 0.19.3
- Platform: Linux-5.15.109+-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version (GPU?): 2.0.1+cu118 (False)
- Huggingface_hub version: 0.16.4
- Transformers version: not installed
- Accelerate version: not installed
- xFormers version: not installed

### Who can help?

@sayakpaul @DN6

**Comment by jloveric** [comment]
Thanks!  I was missing transformers installation!

**Comment by patrickvonplaten** [comment]
Fixed in #4714
